src/server.h

```diff
@@ -1300,6 +1300,8 @@ typedef struct client {
     time_t ctime;           /* Client creation time. */
     long duration;          /* Current command duration. Used for measuring latency of blocking/non-blocking cmds */
     int slot;               /* The slot the client is executing against. Set to -1 if no slot is being used */
+    int cluster_compatibility_check_slot; /* The slot the client is executing against for cluster compatibility check.
+                                           * Set to -2 if we don't check slot, -1 if no slot is being used */
```

Comment:
i was a bit confused at first.
so -1 means we're looking for the slot number and didn't find it yet.
any positive number means we found a slot and no conflicts yet.
and -2 means we already found a violation and reported it?
maybe rephrase.

Reply:
yes, it is a little confused, this variable has two meanings. Actually two variables can make it more clear: 

-  should we check cross slot, maybe we don't need to sample, or we already detect cross slot and don't need to continue (that is useful for script)
- the slot id calculated from the command

but i don't want to introduce two variables to enlarge the client struct, so just use one variable